### DISCUSSION TITLE
Example of function usage

### DISCUSSION THREAD

**Discussion by varac** [getgauge/taiko#2436] [General]
Hi,
I want to use a function in order to de-duplicate code. I'm pretty JS illiterate, but until now I got my way around with taiko.
But I can't get any function to work. My example it: 

```
const { openBrowser, goto, write, click, closeBrowser } = require('taiko');

(async () => {
    try {
        function enter_search_term(term) {
          write(term);
        }
        await openBrowser();
        await goto("duckduckgo.com");
        //await write("test")
        await enter_search_term('Does it work?')
        await press('Enter')
    } catch (error) {
            console.error(error);
    } finally {
            closeBrowser();
    }
})();
```

Is just stalls after entering the term `Does it work?` in the search bar.
I already tried learning asynchronous funktions but couldn't get the above example to work neither. So any help would be appreciated, as well as an documentation update which includes an example containing the use of a function. Thanks!

**Comment by zabil** [comment]
You'll need to define `async` functions and `await` every call of the Taiko API. 
For example the following should work.

```
const { openBrowser, goto, write, click, closeBrowser } = require('taiko');

(async () => {
    try {
        async function enter_search_term(term) {
          await write(term);
        }
        await openBrowser();
        await goto("duckduckgo.com");
        //await write("test")
        await enter_search_term('Does it work?')
        await press('Enter')
    } catch (error) {
            console.error(error);
    } finally {
            closeBrowser();
    }
})();

```

**Reply by varac** [reply]
@zabil Awesome, it works, thanks!
